### PR TITLE
feat: add metadata query parallelization

### PR DESCRIPTION
**Pull request by owczr**
## 📝 Description

**Related Issue:** 
<https://ibm-watsonx.atlassian.net/browse/WXAI-6696>

### 💡 Summary of Changes

This PR parallelizes the metadata queries. Before, they were performed sequentially over a list of tables, which for large databases could take a while. Our investigation had shown that this simple parallelization can speed up AutoAI RAG experiments.

-

---

## ✅ PR Checklist

- [x] **PR Title Format:** `{TYPE}({SCOPE}): {DESCRIPTION}`
  - **Examples:**
    - feat(langchain-ibm): add multi-tenant support  
    - fix(langchain-db2): resolve flag parsing error  
    - docs(langchain-ibm): update API usage examples  
  - **Allowed `{TYPE}` values:**  
    `feat`, `fix`, `docs`, `style`, `refactor`, `perf`, `test`, `build`, `ci`, `chore`, `revert`, `release`
  - **Allowed `{SCOPE}` values (optional):**  
    `langchain-ibm`, `langchain-db2`

- [x] **PR Description:** The *Description* section clearly lists what was changed, why, and how it was tested.

- [x] **Formatting, Linting & Tests**  
  Run the following commands from the root of the modified package(s):

  ```bash
  make format
  make lint
  make test
  ```

  ⚠️ PRs will only be considered if all checks pass in CI.  
  See the [contribution guidelines](https://docs.langchain.com/oss/python/contributing) for more details.

---

## 🧪 Testing (optional)

- Added new unit tests.
- AutoAI RAG works as expected with this change.

---

## 🗒️ Notes (optional)

Thank you again for helping improve **LangChain-IBM**! 🚀  
Your contribution makes the project better for everyone.

### CODE REVIEW

File: libs/ibm/langchain_ibm/utilities/sql_database.py [hunk 1]
```diff
@@ -298,6 +300,7 @@ def __init__(
         include_tables: list[str] | None = None,
         sample_rows_in_table_info: int = 3,
         max_string_length: int = 300,
+        max_workers: int | None = 1,
```
Comment (KarolZm) [inline libs/ibm/langchain_ibm/utilities/sql_database.py:303]: Why `None` is allowed? What if `None`?

Comment (owczr) [reply libs/ibm/langchain_ibm/utilities/sql_database.py:303]: None will use the ThreadPoolExecutor default, should i mention it in the docstring?

Comment (owczr) [reply libs/ibm/langchain_ibm/utilities/sql_database.py:303]: depending on the version it's something like 
> Changed in version 3.13: Default value of max_workers is changed to min(32, (os.process_cpu_count() or 1) + 4).

Comment (KarolZm) [reply libs/ibm/langchain_ibm/utilities/sql_database.py:303]: Yes, I think it's worth to mention some general information. Also, are you able to add unit test with `None` scenario?

Comment (owczr) [reply libs/ibm/langchain_ibm/utilities/sql_database.py:303]: added

File: libs/ibm/langchain_ibm/utilities/sql_database.py [hunk 2]
```diff
@@ -448,19 +453,36 @@ def _check_with_username(
...
+            self._meta_all_tables = {}
+            with ThreadPoolExecutor(max_workers=self._max_workers) as executor:
+                future_to_table = {
+                    executor.submit(fetch_table_metadata, table_name): table_name
+                    for table_name in tables_to_fetch
+                }
+
+                for future in as_completed(future_to_table):
+                    table_name, metadata = future.result()
+                    self._meta_all_tables[table_name] = metadata
```
Comment (MateuszOssGit) [inline libs/ibm/langchain_ibm/utilities/sql_database.py:484]: It is not required to make a try catch block?

Comment (MateuszOssGit) [reply libs/ibm/langchain_ibm/utilities/sql_database.py:484]: Comment resolved in private thread. Closing!

**Review by MateuszOssGit — APPROVED** [review]

**Review by KarolZm — APPROVED** [review]
LGTM
